### PR TITLE
Add ignore_paths input + edit badge check

### PR DESCRIPTION
**Pull request by huiwengoh**
- Add `ignore_paths` input that allows users to specify paths at which checks will be skipped
- Edit google colab badge check to check for html tag
- Add documentation for input variables + check functions

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+description: Lint jupyter notebooks to ensure proper formatting
 
 inputs:
   directory:
-    description: "directory"
+    description: Directory to search for notebooks
     required: false
     default: '.'
+  ignore_paths:
+    description: Comma seperate list of paths to ignore from checks
```
Comment (anishathalye) [inline action.yml]: Grammar nit:

```suggestion
    description: Comma-seperated list of paths to ignore
```

File: action.yml [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
     required: false
     default: true
   check_correct_kernel:
-    description: "TODO"
+    description: Check that notebook is using the right kernel
     required: false
     default: false
   check_notebook_badge:
-    description: "TODO"
+    description: Check that notebooks has a google colab badge
```
Comment (anishathalye) [inline action.yml]: Another grammar nit

```suggestion
    description: Check that notebook has a google colab badge 
```

File: src/nblint.py [hunk 3]
```diff
@@ -22,13 +24,14 @@ def main():
         check(notebook, checks)
 
 
-def find_notebooks(dir):
+def find_notebooks(dir, ignore_paths):
     notebooks = set()
-    for dirname, _, filenames in os.walk(dir):
+    for rootname, dirnames, filenames in os.walk(dir, topdown=True):
```
Comment (anishathalye) [inline src/nblint.py]: I'd suggest sticking to the conventional `dirpath, dirnames, filenames`, matching the Python docs.

**Review by anishathalye — COMMENTED** [review]
I think the semantics of `ignore_path` as they're implemented in this PR are a little confusing. It takes a list of directory names (bare directory names, not paths relative to the `directory` parameter), and skips any directories with those names. So for example, if there are directories `foo/bar` and `baz/bar`, there is no way to skip one but not the other (if you specify "bar", it'll skip both). Furthermore, there isn't a way to skip a particular _file_, which seems like it would also be nice to support.

I'd suggest implementing behavior such that each path in `ignore_paths` is a path or path prefix, and we skip any file matching that path (where the file path is relative to the `directory:` parameter).

In terms of implementation, this should be straightforward. Don't bother with the `topdown=True` / mutating `dirnames`, and for each file, check if `any(os.path.relpath(os.path.join(dirpath, filename), directory).startswith(ignore_path) for ignore_path in ignore_paths)`.

File: src/nblint.py (hunk 3 above)
Comment (anishathalye) [reply src/nblint.py]: Or `dirpath, _, filenames` if we don't need that second element of the tuple

**Comment by huiwengoh** [comment]
Checking this relative path is definitely much clearer than what I had implemented, thanks for the feedback!

**Review by anishathalye — APPROVED** [review]

**Pull request merged**